fortnite_api/shop.py

```diff
@@ -1,266 +1,463 @@
-        self.images: ShopMaterialInstanceImages = ShopMaterialInstanceImages(data['images'])
-        self.colors: Optional[dict] = data['colors']
-        self.scalings: Optional[dict] = data['scalings']
-        self.flags: Optional[dict] = data['flags']
-        self.raw_data: dict = data
-
-
-class ShopMaterialInstanceImages:
-
-    def __init__(self, data: dict):
-        self.offer_image_url: str = data['OfferImage']
-        self.fnm_texture_url: Optional[str] = data.get('FNMTexture')
-        self.image_background_url: Optional[str] = data.get('ImageBackground')
-        self.background_url: Optional[str] = data.get('Background')
-        self.raw_data: dict = data
-
-
-class BrShop:
-    """Represents a Battle Royale shop.
+"""
+MIT License
+
+Copyright (c) 2019-present Luc1412
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from __future__ import annotations
+
+import datetime
+import re
+from typing import Any, Dict, Generic, List, Mapping, Optional, Tuple, Type
+
+from typing_extensions import Self
+
+from .abc import Hashable
+from .asset import Asset
+from .cosmetics import CosmeticBr, CosmeticCar, CosmeticInstrument, CosmeticLegoKit, CosmeticTrack
+from .enums import BannerIntensity
+from .http import HTTPClientT
+from .material import MaterialInstance
+from .proxies import TransformerListProxy
+from .utils import get_with_fallback, parse_time
+
+__all__: Tuple[str, ...] = (
+    "ShopEntryBundle",
+    "ShopEntryBanner",
+    "ShopEntryLayout",
+    "ShopEntryNewDisplayAsset",
+    "ShopEntry",
+    "Shop",
+    "TileSize",
+)
+
+# Matches a tile size: size_<int>_x_<int>
+_TILE_SIZE_REGEX = re.compile(r"size_(?P<width>\d+)_x_(?P<height>\d+)", re.IGNORECASE)
+
+_BACKUP_TILE_MAPPING: Mapping[str, Tuple[int, int]] = {
+    "mini": (1, 1),
+    "small": (1, 1),
+    "normal": (1, 2),
+    "doublewide": (2, 2),
+    "triplewide": (3, 2),
+    "quadwide": (4, 2),
+    "pentawide": (5, 2),
+}
```

Comment:
I think we can remove the backup tile mapping since Epic only uses the new sizing